crates/bevy_ecs/src/event.rs

```diff
@@ -207,6 +207,15 @@ impl<'w, 's, E: Event> EventReader<'w, 's, E> {
         self.reader.len(&self.events)
     }
 
+    /// Retrieves the next event in the [`EventReader`] that has not been seen.
+    ///
+    /// Updates the [`EventReader`]'s event counter, meaning subsequent reads
+    /// will start after the fetched event, if it exists.
+    #[allow(clippy::should_implement_trait)]
+    pub fn next(&mut self) -> Option<&E> {
```

Comment:
`#[allow(clippy::should_implement_trait)]` prevents warning suggesting that `EventReader<E>` should implement `Iterator`. `next()` feels like to appropriate name of this function given that it is "consuming" the _next_ event.